sphinx/writers/latex.py

```diff
@@ -1844,37 +1844,44 @@ def visit_figure(self, node):
                 self.body.append('\\capstart')
             self.context.append(ids + '\\end{sphinxfigure-in-table}\\relax\n')
         elif node.get('align', '') in ('left', 'right'):
-            length = None
-            if 'width' in node:
-                length = self.latex_image_length(node['width'])
-            elif 'width' in node[0]:
-                length = self.latex_image_length(node[0]['width'])
-            self.body.append('\\begin{wrapfigure}{%s}{%s}\n\\centering' %
-                             (node['align'] == 'right' and 'r' or 'l', length or '0pt'))
-            self.context.append(ids + '\\end{wrapfigure}\n')
-        elif self.in_minipage:
-            if ('align' not in node.attributes or
-                    node.attributes['align'] == 'center'):
-                self.body.append('\n\\begin{center}')
-                self.context.append('\\end{center}\n')
+            if self.builder.config.latex_use_wrapfig:
+                length = None
+                if 'width' in node:
+                    length = self.latex_image_length(node['width'])
+                elif 'width' in node[0]:
+                    length = self.latex_image_length(node[0]['width'])
+                self.body.append('\\begin{wrapfigure}{%s}{%s}\n\\centering' %
+                                 (node['align'] == 'right' and 'r' or 'l',
+                                  length or '0pt'))
+                self.context.append(ids + '\\end{wrapfigure}\n')
+
+                # emit an information about wrapfig usage (refs: #3289)
+                logger.info(__('A figure requiring left or right alignment '
+                               'has been detected. LaTeX uses wrapfig macros '
+                               '(cf. ``latex_use_wrapfig`` config variable), '
+                               'please check layout in generated PDF.'),
+                            location=node, type='latex')
```

Comment:
I have replaced warning by less intrusive info. In my testing it does not report source file and line which limits usability. Is there some user option? (time I read our docs ``:)`` )

Reply:
At current implementation, info message does not have location. So `location` argument is ignored.